datalad/log.py

```diff
@@ -457,13 +457,12 @@ def filter(self, record):
         #  logging.Formatter('%(asctime)-15s %(levelname)-6s %(message)s'))
         self.lgr.addHandler(loghandler)
 
-        if is_interactive():
-            phandler = ProgressHandler()
-            # progress only when interactive
-            phandler.addFilter(OnlyProgressLog())
-            # no stream logs of progress messages when interactive
-            loghandler.addFilter(NoProgressLog())
-            self.lgr.addHandler(phandler)
+        phandler = ProgressHandler()
+        # progress only
+        phandler.addFilter(OnlyProgressLog())
+        # no stream logs of progress
+        loghandler.addFilter(NoProgressLog())
+        self.lgr.addHandler(phandler)
```

Comment:
This is problematic.  It means that non-interactively individual `log_progress` calls aren't showing as log messages, but that's a feature at the heart of `log_progress`.

Reply:
I am missing it -- if we get a LogProgressBar, why non-interactive mode should have any special handling here/ in `log_progress`?

Reply:
> if we get a LogProgressBar, why non-interactive mode should have any special handling here/ in `log_progress`?

Because it's a fundamental change to how `log_progess` is intended to work.  It's designed to log _individual_ calls to `log_progress` when operating non-interactively.

As an example, take this script.

```python
import logging
from datalad.log import log_progress

lgr = logging.getLogger("datalad.foo")
nthings = 4

log_progress(lgr.info, "foo",
             "Checking %d things", nthings,
             label="Checking things", total=nthings, unit=" things")
for i in range(nthings):
    log_progress(lgr.info, "foo",
                 "Checking thing %d", i,
                 increment=True, update=1)
log_progress(lgr.info, "foo",
             "Finished checking %d things", nthings)
```

Before this PR, the output is

```
[INFO] Checking 4 things 
[INFO] Checking thing 0 
[INFO] Checking thing 1 
[INFO] Checking thing 2 
[INFO] Checking thing 3 
[INFO] Finished checking 4 things 
```
With this PR, the output is

```
[INFO]  Checking things 4 things done in 0.000181913 sec at 21988 things/sec 
```

This change is entirely disabling the non-interactive side of `log_progress`.  I don't think that's a good thing, but if that's the intent, then the `log_progress` functionality/wrapper should be substantially reworked and reduced if not removed.  And the commit messages should very clearly state this and argue for this change.